src/app/Dashboard/AddCard.tsx

```diff
@@ -0,0 +1,145 @@
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+import * as React from 'react';
+import {
+  Bullseye,
+  Button,
+  Card,
+  EmptyState,
+  EmptyStateBody,
+  EmptyStateIcon,
+  EmptyStateVariant,
+  Select,
+  SelectOption,
+  Stack,
+  StackItem,
+  Text,
+  Title,
+} from '@patternfly/react-core';
+import { Wizard, WizardStep } from '@patternfly/react-core/dist/js/next';
+import { PlusCircleIcon } from '@patternfly/react-icons';
+import { useDispatch } from 'react-redux';
+import { StateDispatch } from '@app/Shared/Redux/ReduxStore';
+import { addCardIntent } from '@app/Shared/Redux/DashboardConfigActions';
+import { DashboardCards, getConfigByName, getConfigByTitle } from './Dashboard';
+
+interface AddCardProps {}
+
+export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardProps) => {
+  const [showWizard, setShowWizard] = React.useState(false);
+  const [selection, setSelection] = React.useState(DashboardCards[0].title);
+  const [selectOpen, setSelectOpen] = React.useState(false);
+  const dispatch = useDispatch<StateDispatch>();
+
+  const options = React.useMemo(() => {
+    return [
+      ...DashboardCards.map((choice, idx) => (
+        <SelectOption key={idx} value={choice.title} description={choice.description}>
+          {choice.title}
+        </SelectOption>
+      )),
+    ];
+  }, [DashboardCards]);
+
+  const handleToggle = React.useCallback(
+    (isOpen) => {
+      setSelectOpen(isOpen);
+    },
+    [setSelectOpen]
+  );
+
+  const handleSelect = React.useCallback(
+    (_, selection) => {
+      setSelection(selection);
+      setSelectOpen(false);
+    },
+    [setSelection, setSelectOpen]
+  );
+
+  const handleAdd = React.useCallback(() => {
+    setShowWizard(false);
+    dispatch(addCardIntent(getConfigByTitle(selection).component.name));
+  }, [dispatch, selection]);
+
+  const handleStart = React.useCallback(() => {
+    setShowWizard(true);
+  }, [setShowWizard]);
+
+  const handleStop = React.useCallback(() => {
+    setShowWizard(false);
+  }, [setShowWizard]);
+
+  return (
+    <>
+      <Card isRounded isLarge>
+        {showWizard ? (
+          <Wizard onClose={handleStop} onSave={handleAdd}>
+            <WizardStep id="card-type-select" name="Card Type" footer={{ nextButtonText: 'Finish' }}>
+              <Stack hasGutter>
```

Comment:
For better visual and following guideline, do you think we should make this content into a form? Then, we can have some sort of form title?

Reply:
I could do it just so that the selection box and description are styled like they would be within a form context but this isn't really a form, so I don't know if that's necessary. I can try it and see what it looks like I guess.

Reply:
The `FormSelectOption` component doesn't seem to have a nice way to include the short description within the `Select`, and putting a `Form` within the `WizardStep` seems to make the container scrollable... I'm not sure this is worth the effort.

Reply:
Ahh I was thinking of something like this:

![Screenshot from 2022-12-09 15-52-54](https://user-images.githubusercontent.com/68053619/206794249-1905c95b-0b21-4239-9f04-9e195809ba58.png)

Hmm now that you mention, its a little too extra?

Reply:
https://www.patternfly.org/v4/components/wizard/design-guidelines

I guess the Wizard example does state that there should be an additional title, which may just repeat what the step title is. I think that's maybe enough.

Right now the wizard could easily just be a form instead, but once there's a Step 2 of configuring the card then the configuration options shown will be specific to each card type and so I think a wizard is the right design pattern.

Reply:
I am trying these changes and looks pretty much achieving the result:

```diff
diff --git a/src/app/Dashboard/AddCard.tsx b/src/app/Dashboard/AddCard.tsx
index c0556e9..28a1c05 100644
--- a/src/app/Dashboard/AddCard.tsx
+++ b/src/app/Dashboard/AddCard.tsx
@@ -44,6 +44,8 @@ import {
   EmptyStateBody,
   EmptyStateIcon,
   EmptyStateVariant,
+  Form,
+  FormGroup,
   Select,
   SelectOption,
   Stack,
@@ -110,16 +112,25 @@ export const AddCard: React.FunctionComponent<AddCardProps> = (props: AddCardPro
         {showWizard ? (
           <Wizard onClose={handleStop} onSave={handleAdd}>
             <WizardStep id="card-type-select" name="Card Type" footer={{ nextButtonText: 'Finish' }}>
-              <Stack hasGutter>
-                <StackItem>
-                  <Select onToggle={handleToggle} isOpen={selectOpen} onSelect={handleSelect} selections={selection}>
-                    {options}
-                  </Select>
-                </StackItem>
-                <StackItem>
-                  <Text>{getConfigByTitle(selection).descriptionFull}</Text>
-                </StackItem>
-              </Stack>
+              <Form>
+                <FormGroup label="Select a card type" isRequired>
+                  <Stack hasGutter>
+                    <StackItem>
+                      <Select
+                        onToggle={handleToggle}
+                        isOpen={selectOpen}
+                        onSelect={handleSelect}
+                        selections={selection}
+                      >
+                        {options}
+                      </Select>
+                    </StackItem>
+                    <StackItem>
+                      <Text>{getConfigByTitle(selection).descriptionFull}</Text>
+                    </StackItem>
+                  </Stack>
+                </FormGroup>
+              </Form>
             </WizardStep>
           </Wizard>
         ) : (
```

Reply:
That looks a lot like I tried and got the weird scrolling behaviour. I'll try that again, maybe I did something a little wrong/different.

Reply:
Oh - I removed the stack, you have it within the form. Maybe that's what the difference was.